examples/text_labels_and_annotations/demo_text_rotation_mode.py

```diff
@@ -17,62 +17,69 @@
   the bounding box of the rotated text.
 - ``roation_mode='anchor'`` aligns the unrotated text and then rotates the
   text around the point of alignment.
-
 """
+
 import matplotlib.pyplot as plt
-from mpl_toolkits.axes_grid1.axes_grid import ImageGrid
+import numpy as np
 
 
 def test_rotation_mode(fig, mode, subplot_location):
     ha_list = ["left", "center", "right"]
     va_list = ["top", "center", "baseline", "bottom"]
-    grid = ImageGrid(fig, subplot_location,
-                     nrows_ncols=(len(va_list), len(ha_list)),
-                     share_all=True, aspect=True, cbar_mode=None)
+    axs = np.empty((len(va_list), len(ha_list)), object)
+    gs = subplot_location.subgridspec(*axs.shape)
+    axs[0, 0] = fig.add_subplot(gs[0, 0])
```

Comment:
this part would be slightly easier with something like https://github.com/matplotlib/matplotlib/pull/14421 available.